common/archive.cpp

```diff
@@ -356,6 +362,24 @@ SeekableReadStream *SearchSet::createReadStreamForMember(const Path &path) const
 	return nullptr;
 }
 
+SeekableReadStream *SearchSet::createReadStreamForMemberNext(const Path &path, const Archive *starting) const {
+	if (path.empty())
+		return nullptr;
+
+	ArchiveNodeList::const_iterator it = _list.begin();
+	for (; it != _list.end(); ++it)
+		if (it->_arc == starting) {
+			++it;
+			break;
+		}
+	for (; it != _list.end(); ++it) {
+		SeekableReadStream *stream = it->_arc->createReadStreamForMember(path);
+		if (stream)
+			return stream;
```

Comment:
I prefer:

```c++
if (SeekableReadStream *stream = it->_arc->createReadStreamForMember(path))
    return stream;
```

Reply:
plain pointer doesn't have any destruction semantics. So I'd prefer to avoid declaring it inside of if condition.

Reply:
I find it clearer with initialization out of the if clause too.
With this style, two things are done as once in the line: assignment and check.

Reply:
Ok